sqlmesh/core/snapshot/evaluator.py

```diff
@@ -450,7 +469,11 @@ def apply(query_or_df: QueryOrDF, index: int = 0) -> None:
 
         with self.adapter.transaction(), self.adapter.session():
             wap_id: t.Optional[str] = None
-            if table_name and snapshot.is_materialized and self.adapter.wap_supported(table_name):
+            if (
+                table_name
+                and snapshot.is_materialized
+                and (model.wap_supported or self.adapter.wap_supported(table_name))
```

Comment:
One tricky thing is what about engines like Trino or Databricks. Technically I could create Iceberg tables but Databricks wouldn't work since it relies on having a Spark Session which isn't always true for Databricks and obviously not implemented at all on Trino.

Reply:
It would still work if the format is set to `iceberg` explicitly, which I think would be required anyway in case of Databricks.

Reply:
Updated databricks